indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,359 @@
+                        <div class="results-list-container" id="location-room-{{ field.id }}-results"></div>
+                    </div>
+                </div>
+            </div>
+            <input id="location-inheritance-{{ field.id }}"
+                   name="location-inheritance-{{ field.name }}"
+                   type="checkbox">
+            <label for="location-inheritance-{{ field.id }}">{% trans %}Use default{% endtrans %}</label>
+            <textarea id="location-address-{{ field.id }}" name="location-address"
+                      placeholder="{% trans %}Address{% endtrans %}"></textarea>
+        </div>
+    </div>
+    <div class="clearfix"></div>
+{% endblock %}
+
+{% block javascript %}
+    <script>
+        (function() {
+            'use strict';
+
+            var field = $('#{{ field.id }}');
+            var description = field.closest('.form-field').find('.form-field-description');
+            var venueInput = $('#location-venue-{{ field.id }}');
+            var roomInput = $('#location-room-{{ field.id }}');
+            var address = $('#location-address-{{ field.id }}');
+            var usingDefault = $('#location-inheritance-{{ field.id }}');
+            var venues = {{ venues | tojson }};
+            var venueNames = {{ venue_names | tojson }};
+            var rooms = {{ rooms | tojson }};
+            var hiddenData = field.val() ? JSON.parse(field.val()) : {};
+            var latestUsedField = '';
+            var latestSelectedOption = {field: '', value: ''};
+
+            if (hiddenData.inheriting) {
+                usingDefault.prop('checked', hiddenData.inheriting);
+            }
+            if (hiddenData.address) {
+                address.val(hiddenData.address);
+            }
+            if (hiddenData.room_name) {
+                roomInput.val(hiddenData.room_name);
+            }
+            if (hiddenData.venue_name) {
+                venueInput.val(hiddenData.venue_name);
+            }
+
+            function toggleDefaultLocation(isChecked) {
+                var message = ('Using default location from: ' + {{ parent[1] | tojson }} +
+                               ' (' + {{ parent[0] | tojson }} + ')');
+                var text = isChecked ? message : 'Using custom location';
+                description.html(text);
+                venueInput.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+                roomInput.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+                address.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+
+                hiddenData.inheriting = isChecked;
+                field.val(JSON.stringify(hiddenData));
+            };
+
+            function reorderCategories(item) {
+                var groupOrder = [item.name];
+                for (var i = venueNames.length - 1; i >= 0; i--) {
+                    if (-groupOrder.indexOf(venueNames[i])) {
```

Comment:
I want to check if an object **is not in** `groupOrder`. Since `indexOf` returns `-1` if the value to search for never occurs, I just added the `-` to invert the `-1` and get `True` in case the object is missing from the array.